presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -223,6 +229,15 @@ private static void sendChallenge(HttpServletResponse response, boolean includeR
         response.setHeader(HttpHeaders.WWW_AUTHENTICATE, formatAuthenticationHeader(includeRealm, Optional.empty()));
     }
 
+    private static void skipRequestBody(HttpServletRequest request)
+            throws IOException
+    {
+        long contentLength = request.getContentLengthLong();
```

Comment:
According to the Javadoc for the method, it can return -1 if the length is not known, so simply discarding everything seems safer.